File: leets/rb_practice/two_sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    output = []
    
    nums.each_with_index do |n, i|
        diff = target - n
        diff_i = nums.index(diff)
        
         if ((nums.include?(diff)) && (i != diff_i))
             output.push(i, diff_i)
             return output
         end
        
    end
end

File: leets/py_practice/lists/twoSum.py
from typing import List

def twoSum(self, nums: List[int], target: int) -> List[int]:
    d = {}
    for i in range(0, len(nums)):
      n = nums[i]
      m = target - n
      if m in d:
          return [d[m], i]
      else:
          d[n] = i
